File: ChallengeAlpha/app/src/main/java/com/hurb/challengealpha/model/Price.kt
package com.hurb.challengealpha.model

import com.google.gson.annotations.SerializedName

data class Price(

    @SerializedName("amount") val amount: Double,
    @SerializedName("old_price", alternate = ["oldPrice"]) val old_price: Double,
    @SerializedName("currency") val currency: String,
    @SerializedName("currency_original") val currency_original: String,
    @SerializedName("gain") val gain: Int,
    @SerializedName("fee_extra_original") val fee_extra_original: Int,
    @SerializedName("gain_original") val gain_original: Int,
    @SerializedName("tariff_policies") val tariff_policies: List<String>,
    @SerializedName("current_price", alternate = ["currentPrice"]) val current_price: Double,
    @SerializedName("total_price") val total_price: Double,
    @SerializedName("fee_extra") val fee_extra: Int,
    @SerializedName("sku") val sku: String,
    @SerializedName("taxes") val taxes: List<Tax>,
    @SerializedName("originalAmountPerDay") val originalAmountPerDay: Double,
    @SerializedName("amountPerDay") val amountPerDay: Double
)

File: DesafioHurb/Domain/Model/Price.swift
//
//  Price.swift
//  DesafioHurb
//
//  Created by Edson Aparecido Guido on 13/09/20.
//  Copyright © 2020 Edson Aparecido Guido. All rights reserved.
//

import Foundation

struct Price: Codable, Equatable {
    let amount: Double
    let priceOldPrice: Double?
    let currency: String?
    let currencyOriginal: String?
    let gain: Int?
    let feeExtraOriginal: Int?
    let gainOriginal: Int?
    let tariffPolicies: [String]?
    let priceCurrentPrice: Double?
    let totalPrice: Double?
    let feeExtra: Int?
    let sku: String
    let taxes: [Tax]?
    let originalAmountPerDay: Double?
    let amountPerDay: Double
    let oldPrice: Int?
    let currentPrice: Int?
    let originalAmount: Int?

    enum CodingKeys: String, CodingKey {
        case amount
        case priceOldPrice = "old_price"
        case currency
        case currencyOriginal = "currency_original"
        case gain
        case feeExtraOriginal = "fee_extra_original"
        case gainOriginal = "gain_original"
        case tariffPolicies = "tariff_policies"
        case priceCurrentPrice = "current_price"
        case totalPrice = "total_price"
        case feeExtra = "fee_extra"
        case sku, taxes, originalAmountPerDay, amountPerDay, oldPrice, currentPrice, originalAmount
    }
}
